tools/scripts/composer/pear-execption-fix.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+## Cleanup the vendor tree. The main issue here is that civi Civi is
+## deployed as a module inside a CMS, so all its source-code gets published.
+## Some libraries distribute admin tools and sample files which should not
+## be published.
+##
+## This script should be idempotent -- if you rerun it several times, it
+## should always produce the same post-condition.
+
+## Replace a line in a file
+## This is a bit like 'sed -i', but dumber and more cross-platform.
+function simple_replace() {
+  php -r 'file_put_contents($argv[1], preg_replace($argv[2], $argv[3], file_get_contents($argv[1])));' "$@"
+}
+
+
+# add in class_exists test as per CRM-8921.
+simple_replace vendor/pear/pear_exception/PEAR/Exception.php '^\<\?php^' '<?php
+  // CRM-8921
+  if ( class_exists( "'"PEAR_Exception"'" ) ) {
+     return; 
+  }' /home/seamus/test.php
```

Comment:
I don't think this reaches the idempotency. To see this, simply run the script multiple times -- it will repeatedly add the clause.

Suggestion: maybe search for "CRM-8921" as a key to decide whether to patch. It seems highly unlikely that upstream would organically include that text...

```
if ! grep -q CRM-8921 vendor/pear/pear_exception/PEAR/Exception.php ; then 
  simple_replace ...
; fi
```

Totally random suggestion: in this scenario, patchability might be more important than code-style, so maybe write it in a way that injects one physical line (that can be easily found/replaced via regex), e.g.

```
if (class_exists('PEAR_Exception')) return; /* CRM-8921 */
```